test/Servant/QuickCheck/InternalSpec.hs

```diff
@@ -107,6 +112,17 @@ isComprehensiveSpec = describe "HasGenRequest" $ do
     let _g = genRequest comprehensiveAPIWithoutRaw
     True `shouldBe` True -- This is a type-level check
 
+queryParamsSpec :: Spec
+queryParamsSpec = describe "QueryParams" $ do
+
+  it "reduce to an HTTP query string correctly" $ do
+    let rng = mkQCGen 0
+        burl = BaseUrl Http "localhost" 80 ""
+        gen = genRequest paramsAPI
+        req = (unGen gen rng 0) burl
+        qs = C.unpack $ queryString req
+    qs `shouldContain` ("one=")
```

Comment:
I think this test doesn't check what I'm concerned about above. Some suggestions below.